lib/active_merchant/billing/gateways/mondido.rb

```diff
@@ -0,0 +1,658 @@
+        #     :state => '',
+        #     :country => '',
+        #     :zip => '90210',
+        #     :phone => ''
+        #   }
+        # }
+        ## There are 3 different addresses you can use.
+        ## There are :billing_address, :shipping_address, or you can just pass in
+        ## :address and it will be used for both.
+
+        requires!(options, :order_id)
+        options[:order_id] = options[:order_id].to_s
+
+        post = {
+          # string* required
+          # The ID of the merchant
+          :merchant_id => @merchant_id,
+
+          # decimal* required
+          # The transaction amount ex. 12.00
+          :amount => get_amount(money, options),
+
+          # string* required
+          # Merchant order/payment ID
+          :payment_ref => options[:order_id],
+
+          # string* required
+          # The currency (SEK, CAD, CNY, COP, CZK, DKK, HKD, HUF, ISK, INR, ILS, JPY, KES, KRW,
+          #  KWD, LVL, MYR, MXN, MAD, OMR, NZD, NOK, PAB, QAR, RUB, SAR, SGD, ZAR, CHF, THB, TTD,
+          #  AED, GBP, USD, TWD, VEF, RON, TRY, EUR, UAH, PLN, BRL)
+          :currency => get_currency(money, options),
+
+          # string * required
+          # The hash is a MD5 encoded string with some of your merchant and order specific parameters,
+          # which is used to verify the payment, and make sure that it is not altered in any way.
+          :hash => transaction_hash_for(money, options),
+        }
+
+        ## API Optional Parameters
+        #
+        # - test
+        # - authorize
+        # - metadata
+        # - store_card
+        # - plan_id
+        # - customer_ref
+        # - webhook
+        # - process
+
+        # test (boolean)
+        #   Whether the transaction is a test transaction. Defaults false
+        post[:test] = test?
+
+        # authorize (boolean)
+        #   [ Not documented ]
+        post.merge!( :authorize => options[:authorize] ) if options[:authorize]
+
+        # Merchant custom Metadata (string)
+        #   Metadata is custom schemaless information that you can choose to send in to Mondido.
+        #   It can be information about the customer, the product or about campaigns or offers.
+        #
+        #   The metadata can be used to customize your hosted payment window or sending personalized
+        #   receipts to your customers in a webhook.
+        #
+        #   Details: http://doc.mondido.com/api#metadata
+        if options[:metadata]
+          if options[:metadata].is_a? Hash
+            options[:metadata] = options[:metadata].to_json
+          end
+          post.merge!( :metadata => options[:metadata] ) 
+        end
+
+        # Store Card (boolean)
+        #   true/false if you want to store the card
+        post.merge!( :store_card => options[:store_card] ) if options[:store_card]
+
+        # Plan ID (int)
+        #   The ID of the subscription plan.
+        post.merge!( :plan_id => options[:plan_id] ) if options[:plan_id]
```

Comment:
Please remove all functionality related to subscriptions/recurring. They're outside of scope for ActiveMerchant.